### ISSUE TITLE
Blank overlay

### ISSUE THREAD

**Issue by sp00kycz** [Alia5/GloSC#33]
I have a problem with overlay, which remains empty only with GloSC logo in the middle. Does not matter how I set up the app shortcuts, the result is the same.

Controller itself, its bindings and properties are working properly. I can use touch menu/wheel menu and all other functions and even vibration feedback is great, they are just not visible on screen.
I already tried to completely remove GloSC and reinstall it with no result. On my second computer it works flawlessly, the software configuration of these two computers is more or less equal. Only difference is that my main laptop has touchscreen which is high DPI (2160x1440, 12"). Could this be an issue?

Also, the app is not exactly high DPI ready, some of its texts and hints are not displayed correctly, but it does not affect usability.

All in all, awesome app, one of a kind!

**Comment by Alia5** [comment]
I don't think HDPI should have something to do with that.

Make sure the BPM Overlay shows up instead of the "regular" one (Some setting in Steam)
Try opening and closing the overlay using the Steam button

You may also trying to opt-in / -out of Steam client betas

**Comment by sp00kycz** [comment]
So I tried to play with some settings and stuff and this is the result:
- if BPM overlay is enabled, I can access the overlay via the Steam button, yet only GloSC logo is displayed; same result with Shift+TAB when GloSC overlay window is selected
- if BPM overlay is disabled, I can access the overlay via Shift+TAB when GloSC overlay window is selected and it works properly, though I cannot use Steam button to pop the overlay or use the Touch/Wheel menus
- Steam Beta does not have any influence on this behavior
- running shortcuts from Big Picture also does not have any influence on this behavior

**Comment by Alia5** [comment]
Can I send you a special build the coming days which spits me out a giant log file?
Or hop onto discord some time ;)

Not being able to access the overlay when BPM is disabled is a known issue...
Need more time to reverse stuff in the overlay

**Comment by KeyboardSounds** [comment]
I'm also experiencing this issue. Controller works as configured, but I can't see touch menus or anything. I get the GloSC logo when I press the middle steam button. Happy to help with debugging :)

**Comment by Alia5** [comment]
Can you test with this?
https://github.com/Alia5/GloSC/releases/tag/1.4.2

**Comment by sp00kycz** [comment]
Thanks for the update. Unfortunately it doesn't seem to help regarding this issue. Still blank overlay with GloSC logo. I'm still open to your special build to output a log file, if that would help.
Additionaly, I found out that ViGEm is not being installed with 1.4.2 version. I had to install 1.4.1 to get ViGEm and then install 1.4.2 (I hope that's not a problem).
Also, I'm not able to add shortcuts from GloSC if the Steam is run in offline mode, ie. without being connected to Internet. Is that intentional? I thought Steam library is managed on client side, at least non-steam shortcuts, so it shouldn't matter whether I'm online or not. Getting "Couldn't detect Steam shortcuts file! Make sure you have at least one non-Steam shortcut for the file to be present Steam must be running for it to be detected" error message.

**Comment by Alia5** [comment]
That one should fix it.
However, it'll most likely break with new steam updates.
I'm on Steam beta, so you might have to try that.

[SteamTarget.zip](https://github.com/Alia5/GloSC/files/1594568/SteamTarget.zip)

**Comment by sp00kycz** [comment]
Still nothing, at least for me. Don't know about @KeyboardSounds if it's fixed for her.
I'm also on Steam Beta, newest build.
I hope you're not getting too frustrated about this. Fingers crossed!

**Comment by AnoshW** [comment]
I just tried to use the SteamTarget.zip from the comment made on the 30th December. It put the GloSC logo window as always on the top and pressing the steam button didn't close the logo. Still no overlay elements. When the steam button/shift+tab is pressed the mouse course disappears and at the top of the screen, there appears to be a long black bar spanning the full length of the screen.

**Comment by Alia5** [comment]
That SteamTarget attached here won't work anymore ;)
I'm surprised that it didn't work back then... Anyway..

I'd still really like a way to reproduce this.

**Comment by Alia5** [comment]
I've started working on a v2 of GloSC...

You can test a new method of detecting overlay opening / closing with a very much work in progress SteamTarget.
[SteamTarget_v2_TEST.zip](https://github.com/Alia5/GloSC/files/1799991/SteamTarget_v2_TEST.zip)

Don't actually try replacing this with the one from you GloSC installation, **it will _not_ work!**
Just launch that via Steam and open/close the overlay a bunch of times.
You should see corresponding messages popping up in the console window.

**If** not, you may want to try on Steam-Beta!

**Comment by Alia5** [comment]
> overlay only shows up when NVidia settings for the program has been set to GPU (works but with black screen)
A redditor said somethign along those lines a while back
It's even stated in the FAQ.

To help me understand this correctly:
If the NVidia settings are set to GPU, the overlay shows up fine, but it covers your screen in black?
-> That would mean I could "hide" the overlay window completely if it's not shown?


Good to hear that it works, though :)

**Comment by AnoshW** [comment]
Yeah, so say the NVidia setting is set to GPU, the entire screen is black. When the steam button is pressed, the overlay shows up and works fine and then disappears when the steam button is pressed again.
But if the NVidia setting is set to integrated, the overlay window is transparent but, when the steam button is pressed there is no overlay, only the GloSC logo. No touch menus either.
Would hiding the overlay mean the touch menus not show up?

Thanks, keep up the good work :)

**Comment by Alia5** [comment]
> Would hiding the overlay mean the touch menus not show up?
Nope... 😪

Does this issue only apply to devices with NVidia Optimus (eg. notebooks) or can this happen on desktops as well?
Never had an NVidia GPU in years, so it's kinda hard to think about a good solution

**Comment by Alia5** [comment]
Additionally, have a [semi-functional SteamTarget](https://github.com/Alia5/GloSC/files/1800979/Release.zip), freshly built from the v2 branch ;)

It supports controller emulation, forcing Steam to keep bindings, as well as making the overlay clickable.
No workaround/fix for the NVidia issue, though :/

Again, **do not** replace this with the one from you GloSC installation!

Just add it to Steam and go from there.

Small disclaimer:

> It currently cannot launch any games, and it **does not unhook** Steam **if** closed via the SteamOverlay itself ( or killed via Task Manager )

> Either close via the provided GloSC windows, or restart Steam after using this.
I am not responsible for your VAC status if you go and play CS:GO after not following this.

**Comment by Alia5** [comment]
@sp00kycz @AnoshW 

I have here a testing build from the v2 branch.
In addition to the (also in previous comments) new method for detecting overlay opening/closing, you can also find a ```bEnableOverlayOnlyConfig=0``` flag in the .ini file.
This implements the previously mentioned workaround for the overlay.
You sadly won't be able to see Touch-/Radial-Menus, but it's better than nothing for now, I guess.

[GloSC-installer.zip](https://github.com/Alia5/GloSC/files/1822895/GloSC-installer.zip)

Could you reply on how this works for you?

**Comment by AnoshW** [comment]
Works well for me. One thing I have noticed is when GloSC is not the window in the foreground and the steam button is pressed, the overlay comes up after a short delay (~1 second - not really a deal breaker for me) but, the controller doesn't switch to controlling the overlay until you alt-tab to the GlocSC window. Tested this with`bEnableOverlayOnlyConfig` enabled and disabled. Both gave the same result.

Apart from that, works great. If there is anything specific that you'd like me to test, please let me know.

**Comment by JoeMBlair** [comment]
I tested the new build but I still don't see the overlay. I can hear the sounds of it but I can't see it. My laptop only has integrated Intel graphics.

**Comment by AnoshW** [comment]
Yes, forgot to mention the same issues is still present for me too.  Tired `bEnableOverlayOnlyConfig=1`  with integrated graphics and Nvidia GPU but still has the same issue.

**Comment by Alia5** [comment]
> [...] ```bEnableOverlayOnlyConfig=0``` flag in the .ini file.
> This implements the previously mentioned workaround for the overlay.

That one:
> To help me understand this correctly:
If the NVidia settings are set to GPU, the overlay shows up fine, but it covers your screen in black?
-> That would mean I could "hide" the overlay window completely if it's not shown?

Meaning the flag just hides the overlay when it's not shown, where your screen would just be black, and shows the window upon opening.

Does this work? I'm sorry but things are still unclear...

**Comment by AnoshW** [comment]
Hope this quick video clears things up.
https://drive.google.com/file/d/1DGs3C4-2cymy84rAD-jg9J1fUCi8YuRR/view?usp=sharing
Basically still getting the black screen :(

**Comment by Alia5** [comment]
You have to change that parameter in the config file **of your particular shortcut**
They are in the subfolder "targets" (IIRC) and named after your shortcuts...

**Comment by AnoshW** [comment]
Still getting the same black screen after adding the parameters into the shortcut config.

**Comment by Alia5** [comment]
@AnoshW I never fully implemented that flag -.-

It should work now
https://github.com/Alia5/GloSC/releases/tag/2.0.0.117

Additionally, I found an interesting Issue in one of the used libraries. It might not even be necessary anymore(?)

**Comment by AnoshW** [comment]
Works great! Set the NVidia setting to gpu and `bEnableOverlayOnlyConfig=1` otherwise it doesn't work.

I do miss the touch menus though. Is there any hope?

**Comment by Alia5** [comment]
Probably not, sorry.

Ouh BTW, I totally forgot to mention, you have to have DWM enabled and cannot have a legacy system theme, as that would disable, DWM

**Comment by nonplayer** [comment]
I am now getting this same issue on a brand new Dell XPS laptop with nvidia graphics. The game imports fine, but when launching with BPM overlay, it shows no overlay. Additionally, when pressing the steam button, all I get is the Glosc Logo dead center of the screen.
